app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -38,16 +58,60 @@ class MainActivity : ComponentActivity() {
     @Inject
     lateinit var lazyStats: dagger.Lazy<JankStats>
 
+    @Inject
+    lateinit var userDataRepository: UserDataRepository
+
     override fun onCreate(savedInstanceState: Bundle?) {
-        installSplashScreen()
+        val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
+        /**
+         * The current user data, updated here to drive the UI theme
+         */
+        var userDataResult: Result<UserData> by mutableStateOf(Result.Loading)
+
+        // Update the user data
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                userDataRepository.userDataStream
+                    .asResult()
+                    .onEach {
+                        userDataResult = it
+                    }
+                    .collect()
+            }
+        }
+
+        // Keep the splash screen on-screen until the user data is loaded
+        splashScreen.setKeepOnScreenCondition {
+            when (userDataResult) {
+                Result.Loading -> true
+                is Result.Success, is Result.Error -> false
+            }
+        }
+
         // Turn off the decor fitting system windows, which allows us to handle insets,
         // including IME animations
         WindowCompat.setDecorFitsSystemWindows(window, false)
 
         setContent {
-            NiaApp(calculateWindowSizeClass(this))
+            val systemUiController = rememberSystemUiController()
+            val darkTheme = shouldUseDarkTheme(userDataResult)
+
+            // Update the dark content of the system bars to match the theme
+            DisposableEffect(systemUiController, darkTheme) {
+                systemUiController.systemBarsDarkContentEnabled = !darkTheme
+                onDispose {}
```

Comment:
If we don't have anything to dispose, can we use a `LaunchedEffect`?

Reply:
Yes, although I think they are slightly different: From my understanding `DisposableEffect` wlil run slightly sooner than `LaunchedEffect` since `LaunchedEffect` requires a dispatch. I don't know if that is a large enough difference that this call will be delayed by a frame or not.

Reply:
It might not be, the dangling lambda threw me off for a bit. If the preference is for `DisposableEffect`, we should probs add a comment explaining why.

Reply:
Leaving as-is for now.